### PR TITLE
Add resource limits for master resources in density test

### PR DESCRIPTION
**Pull request by gmarek**
cc @jlowdermilk @fabioy

### CODE REVIEW

**Comment by jlowdermilk** [comment]
lgtm

Ref #23820.

File: test/e2e/density.go [hunk 1]
```diff
@@ -50,7 +50,57 @@ const (
...
+	schedulerMem = math.MaxUint64
+	if framework.ProviderIs("kubemark") {
+		if numNodes < 10 {
+			apiserverCPU = 0.15
+			apiserverMem = 150 * (1024 * 1024)
+			controllerCPU = 0.1
+			controllerMem = 100 * (1024 * 1024)
+			schedulerCPU = 0.05
+			schedulerMem = 50 * (1024 * 1024)
+		} else if numNodes < 250 {
```
Comment (wojtek-t) [inline test/e2e/density.go]: As discussed offline - I would change to <=100, <=500, <=1000 ?

Comment (gmarek) [inline test/e2e/density.go]: Done. @wojtek-t PTAL

**Comment by wojtek-t** [comment]
lgtm

**Comment by alex-mohr** [comment]
buildcop says: this seems to make tests better, so marking p2.

**Comment by wojtek-t** [comment]
@k8s-oncall - it seems that this PR broke kubemark-500. We should probably revert this.

**Comment by alex-mohr** [comment]
FYI, revert of this PR merged.

**Pull request merged**